### ISSUE TITLE
Parse for JSON?

### ISSUE THREAD

**Issue by paologaleotti** [Oudwins/zog#55]
Hello, i think it is a bit tricky to actually validate json. If i see correctly, the only accepted type for `Parse()` is `map[string]any`. Would it be possible to also accept something very common like a `json.RawMessage` or the aliased `byte[]`?

Because i want to validate my json but i'm NOT on a http server (which is a common use case) i need to first unmarshal into a struct (so my `json:` struct tags work as expected) and then pass the struct to zog to validate.

So in summary, it would be nice to have the ability to Parse a struct OR some method to parse directly `byte[]`.

**Comment by Oudwins** [comment]
Hey! thanks for creating the issue. There are two things that are planned that I hope will help address this. 

The first is that I agree that zog should provide some way of parsing raw json, but I haven't decided on how this will be done yet.

The second is the Validate method which has been planned for a while but I haven't gotten around to releasing it because it causes some pains with the current mental model. But I should be able to release the first working version of this hopefully over these holidays. That way you can just parse the json into your struct then validate it directly

**Comment by Oudwins** [comment]
@paologaleotti I just wanted to let you know that I have now extracted the code that does the json parsing from the zhttp package into zjson. Which means you can now use that for parsing. I'll also hopefully have completed the Validate method before the holidays are over and that is another option you could use.

**Comment by paologaleotti** [comment]
Amazing thank you!
